src/main/java/com/aws/greengrass/clientdevices/auth/ClientDevicesAuthService.java

```diff
@@ -170,13 +172,28 @@ private int getValidCloudCallQueueSize(Topics topics) {
     @Override
     protected void install() throws InterruptedException {
         super.install();
+
+        // Before subscribing to the config updates, look up for CA type topic. This will create a node with default
+        // value as null if the topic doesn't exist already.
+        getContext().runOnPublishQueueAndWait(() -> {
+            this.config.lookup(CONFIGURATION_CONFIG_KEY, CERTIFICATE_AUTHORITY_TOPIC, CA_TYPE_TOPIC);
+        });
+        getContext().waitForPublishQueueToClear();
+
+        // Subscribe to config updates
+        configChangeHandler();
+
+        // Update CA based on the config.
+        updateCAType();
+    }
+
+    private void configChangeHandler() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.